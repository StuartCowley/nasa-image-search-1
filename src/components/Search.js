import React, { useState } from 'react';
import '../styles/Search.css';
import getImages from '../requests/getImages';
import PropTypes from 'prop-types';

const Search = ({ setSearchResults, setErrorMessage }) => {
	const [value, setValue] = useState();

	const handleSubmit = async event => {
		event.preventDefault();
		const images = await getImages(value);
		setSearchResults(images);
		if (images.length === 0) {
			setErrorMessage(true);
		} else {
			setErrorMessage(false);
		}
	};
	return (
		<div className="search-wrapper">
			<form className="search-form" onSubmit={handleSubmit}>
				<input
					className="search-input"
					type="text"
					placeholder="find your perfect space"
					onChange={event => setValue(event.target.value)}
				/>
				<button className="search-button" type="submit">
					SEARCH IMAGES
				</button>
			</form>
		</div>
	);
};

Search.propTypes = {
	setSearchResults: PropTypes.func,
};

export default Search;
