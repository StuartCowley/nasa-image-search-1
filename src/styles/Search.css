.search-input {
	width: 300px;
	height: 25px;
	padding: 5px;
	outline-color: black;
	border-radius: 5px 0px 0px 5px;
}
.search-button {
	width: 100px;
	height: 40px;
	padding: 5px;
	background-color: white;
	outline-color: black;
	align-items: center;
	border-radius: 0 5px 5px 0;
	cursor: pointer;
}
.search-form {
	display: inline-flex;
	align-items: center;
}
.search-wrapper {
	text-align: center;
	padding: 10px;
}
