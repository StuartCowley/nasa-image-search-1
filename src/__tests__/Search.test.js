import React from 'react';
import { render, screen } from '@testing-library/react';
import Search from '../components/Search';

describe('Search', () => {
	const validProps = {
		setSearchResults: () => {},
	};
	it('renders correctly', () => {
		const { asFragment } = render(
			<Search setSearchResults={validProps.setSearchResults} />,
		);
		expect(asFragment()).toMatchSnapshot();
	});
	it('button renders with correct text', () => {
		render(<Search setSearchResults={validProps.setSearchResults} />);
		const buttonElement = screen.getByText('SEARCH IMAGES');
		expect(buttonElement).toBeInTheDocument();
	});
});
