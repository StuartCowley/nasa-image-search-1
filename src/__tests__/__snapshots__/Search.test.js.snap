// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-wrapper"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        placeholder="find your perfect space"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        SEARCH IMAGES
      </button>
    </form>
  </div>
</DocumentFragment>
`;
